test(ci): Hardstop-Test vergleicht lokales Datum gegen UTC-Datum des Skripts

## Symptom

`tests/scripts/test_check_pip_audit_hardstop.py::test_on_hardcutoff_day_list_must_already_be_empty` schlägt lokal deterministisch fehl, wenn man ihn zwischen 00:00 und 02:00 Europe/Berlin laufen lässt:

```
assert result.returncode == 2
E   AssertionError: assert 0 == 2
E    +  stdout='OK: 2026-08-10 ist vor dem Hardcutoff 2026-08-11 — pip-audit --ignore-vuln ist erlaubt.'
```

## Ursache

Zwei verschiedene Zeitzonen für dasselbe „heute":

- Der Test bildet den Cutoff mit `datetime.date.today().isoformat()` — **lokale** Zeitzone.
- `scripts/check-pip-audit-hardstop.sh` bildet `TODAY="$(date -u +%F)"` — **UTC**.

In Europe/Berlin (UTC+1/+2) sind das zwischen Mitternacht und 01:00 bzw. 02:00 Uhr zwei verschiedene Daten. Der Test setzt dann `PIP_AUDIT_HARDCUTOFF` auf den *Folgetag* aus Sicht des Skripts, das Skript nimmt korrekt den „vor dem Hardcutoff"-Zweig und liefert Exit 0 statt der erwarteten 2.

## Warum das in CI nie auffällt

GitHub-Runner laufen in UTC. Dort sind beide Werte identisch, der Test ist grün — und bleibt es. Der Fehler trifft ausschließlich lokale Läufe im nächtlichen Zeitfenster.

## Erwartetes Verhalten

Der Test prüft die Inklusiv-Semantik am Cutoff-Tag. Diese Absicht ist richtig; nur die Datumsquelle ist es nicht. Test und Skript müssen dieselbe Zeitbasis verwenden — naheliegend ist `datetime.datetime.now(datetime.UTC).date().isoformat()` im Test, damit das Skript die maßgebliche Quelle bleibt.

## Reproduktion

```bash
cd backend
TZ=Europe/Berlin faketime '2026-08-11 00:30:00' uv run pytest \
  tests/scripts/test_check_pip_audit_hardstop.py::test_on_hardcutoff_day_list_must_already_be_empty
```

Ohne `faketime`: im genannten Zeitfenster schlicht ausführen.

## Nebenbefund

Fünf weitere Tests in `tests/test_sync_status_cache.py` scheitern lokal, wenn `backend/.cache` einem anderen Benutzer gehört (typisch nach einem Container-Lauf — der Compose-Stack bindet `./backend/.cache/huggingface`). Das ist ein Umgebungsproblem, kein Testdefekt, und in `docs/STATUS.md` sowie `docs/deployment-dev.md` dokumentiert. Ob die Tests dafür einen sauberen Skip statt eines Fehlschlags liefern sollten, ist eine eigene Frage.

Gefunden beim Doku-Sync in #1202.

--- a/backend/tests/scripts/test_check_pip_audit_hardstop.py
+++ b/backend/tests/scripts/test_check_pip_audit_hardstop.py
@@ -50,8 +50,16 @@ def test_after_hardcutoff_with_ignore_list_fails_with_exit_2() -> None:
 
 def test_on_hardcutoff_day_list_must_already_be_empty() -> None:
     # Inklusiv-Semantik: am Cutoff-Tag selbst ist die Liste nicht mehr erlaubt.
+    #
+    # #1203: ``date.today()`` liefert das Datum der LOKALEN Zeitzone, das
+    # Skript bildet sein "heute" aber mit ``date -u`` (UTC, Zeile 26). In
+    # Europe/Berlin (UTC+1/+2) sind das zwischen Mitternacht und 01:00 bzw.
+    # 02:00 zwei verschiedene Tage: der Test setzt den Cutoff dann auf den
+    # Folgetag aus Sicht des Skripts, das nimmt korrekt den
+    # "vor dem Hardcutoff"-Zweig und liefert Exit 0 statt 2.
+    # Beide Seiten muessen dieselbe Zeitzone verwenden.
     import datetime
 
-    today = datetime.date.today().isoformat()
+    today = datetime.datetime.now(datetime.UTC).date().isoformat()
     result = _run(today, "--ignore-vuln GHSA-xxxx")
     assert result.returncode == 2

--- a/changelog.d/1203-hardstop-utc.md
+++ b/changelog.d/1203-hardstop-utc.md
@@ -1,0 +1,10 @@
+### Behoben
+
+- `test_on_hardcutoff_day_list_must_already_be_empty` verglich das Datum der
+  lokalen Zeitzone (`datetime.date.today()`) gegen das UTC-Datum, das
+  `scripts/check-pip-audit-hardstop.sh` mit `date -u` bildet. In Zeitzonen mit
+  UTC-Versatz fielen beide zwischen Mitternacht und dem Versatz auf
+  verschiedene Tage — der Test setzte den Cutoff dann auf den Folgetag aus
+  Sicht des Skripts, das nahm korrekt den „vor dem Hardcutoff"-Zweig und lieferte
+  Exit 0 statt der erwarteten 2. In Europe/Berlin betraf das das Zeitfenster
+  00:00–02:00. Der Test bildet sein „heute" jetzt ebenfalls in UTC. (#1203)

